azure-iot-device/tests/provisioning/test_sync_provisioning_device_client.py

```diff
@@ -111,10 +123,16 @@ def test_client_register_success_calls_polling_machine_register_with_callback(
         mock_polling_machine_init.return_value = mock_polling_machine_success
 
         client = ProvisioningDeviceClient(mqtt_provisioning_pipeline)
-        client.register()
+        result = client.register()
 
         assert mock_polling_machine_success.register.call_count == 1
         assert callable(mock_polling_machine_success.register.call_args[1]["callback"])
+        assert result is not None
```

Comment:
>assert result is not None [](start = 8, length = 25)

Building on what was said above re: fixtures, once the expected result is brought into the test explicitly as a fixture (say, `success_result`), you can then compare the result directly to the expected result.

You also probably don't need to check all the individual values - if you `assert result is success_result`, you know that the object returned is the same one. This is nice because it also allows us to remove the magic of somehow knowing that the result registration state should be `fake_registration_state`, when, again, from reading just this test, we have no way of knowing what the expected result should be.

Admittedly, though, now that I'm thinking about it, this approach doesn't account for the possibility that the object was modified on the way to being returned....

Reply:
check replies above

---
In reply to: [315808899](https://github.com/Azure/azure-iot-sdk-python-preview/pull/169#discussion_r315808899) [](ancestors = 315808899)